javaslang/src/main/java/javaslang/collection/Seq.java

```diff
@@ -437,7 +437,7 @@ default int lastIndexWhere(Predicate<? super T> predicate) {
      * this sequence in a Some if the index is within bounds, otherwise a None.
      */
     default Function1<Integer, Option<T>> lift() {
-        return i -> (i >= length() || i < 0) ? Option.none() : Option.some(apply(i));
+        return i -> Option.when(i < length() && i >= 0, () -> apply(i));
```

Comment:
Good!
Let's switch the conditions, too: `Option.when(i >= 0 && i < length(), ...)`. Then we omit the `length()` call if `i < 0`.